### PR TITLE
Refactor/firebase tokens

### LINKED ISSUE #50: firebase admin needs to be centralized?
I'm not sure how firebase credentials are doled out in general but any user who needs to be able to publish datasets will be blocked without proper authentication.

Perhaps if we had github actions that could publish data then we could store the creds in a single place here in this repo.

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
closes #50 

Solution
========
What I/we did to solve this problem
- Updated Readme with 3 database options(production, staging, dev) and required tokens in each option 
- Moved part of the credentials out of `.env` into 2 separate files - `dev-creds` and `production-creds`
- Updated token references in `setup-firebase`
  
with @meganrm 

## Type of change
Please delete options that are not relevant.
* Refactor and maintenance 
* This change requires a documentation update

### CODE REVIEW

**Comment by toloudis** [comment]
I have a few questions.  My understanding is that this is going to enable us to ultimately publish datasets to staging via github actions (and maybe via command line) for scientists to be able to see their data earlier.

It looks like, from the README, that the env file is now shorter.  Does anything else change?  

Is "dev" a brand new option now?  Maybe we could have a bit more info about how to set up a dev database? (And is CFE set up to access a dev db or does it require changes?)

File: README.md [hunk 1]
```diff
@@ -2,36 +2,40 @@
...
+DEV_FIREBASE_DB_URL=
 ```
+> To access `AWS`, `production`, or `staging`, please contact the development team for the necessary credentials 
+
 # To process a dataset:
 
 ## Setup
+### Set up a dev database
+- set NODE_ENV="dev"
+- add your required secret tokens to .env file
```
Comment (meganrm) [inline README.md:37]: We should have at least a link to a tutorial on setting up a firebase database. And some information on where to find the particular variable you need and what they're called in firebase

Comment (meganrm) [reply README.md:37]: Imagine a new developer is going to start working on this: we want them to be able to set up the repo and start working just by reading the README and not having to ask you or I how to do it

File: README.md [hunk 2]
```diff
@@ -22,21 +33,16 @@ STAGING_FIREBASE_TOKEN=
...
 
 # used if NODE_ENV === "dev"
-DEV_FIREBASE_TOKEN=
-DEV_FIREBASE_EMAIL=
-DEV_FIREBASE_ID=
-DEV_FIREBASE_DB_URL=
+DEV_FIREBASE_TOKEN= project settings/service accounts/generate new private key
+DEV_FIREBASE_EMAIL= project settings/services accounts/firebase service account 
+DEV_FIREBASE_ID= project settings/general/project ID
+DEV_FIREBASE_DB_URL= "https://`DEV_FIREBASE_ID`.firebaseapp.com"
```
Comment (toloudis) [inline README.md:39]: does this mean that the DEV_FIREBASE_DB_URL can be inferred from the DEV_FIREBASE_ID and is not really needed in the env file?

Comment (rugeli) [reply README.md:39]: oh good catch! You're right we can just infer url in dev-creds file

**Review by toloudis — APPROVED** [review]
Thanks for doing this!  One more step closer to automatic data deployment.  How will github-actions work?  Will there need to be a special .env file stored as a secret in github?

**Review by meganrm — APPROVED** [review]

**Comment by rugeli** [comment]
> Thanks for doing this! One more step closer to automatic data deployment. How will github-actions work? Will there need to be a special .env file stored as a secret in github?

As far as I understand, the next step is to create a new yaml file in `workflows` which can be manually triggered using the new workflow_dispatch event. Shouldn't need to add any secret tokens in .env file.

**Pull request merged**